_includes/snippets_library/UCL_Myriad_sge/14/gcc_example.snip

```diff
@@ -0,0 +1,172 @@
+To be more specific, you can specify the particular software you want.
+e.g.
+
+```
+{{ site.remote.prompt }} module avail stata
+```
+{: .bash}
+ 
+```
+------------------------------------- /shared/ucl/apps/modulefiles/applications --------------------------------------
+stata/14 stata/15
+```
+{: .output}
+
+Let's take a closer look at the `matlab` module.
+Matlab is a widely-used piece of software which uses matrix multiplication.
+As we shall see, there are different versions available, and we want to make sure the one we use
+is the correct one for our purposes.
+
+Let's see which versions we have access to.
+
+```
+{{ site.remote.prompt }} module avail matlab
+```
+{: .bash}
+
+```
+------------------------------------- /shared/ucl/apps/modulefiles/applications --------------------------------------
+matlab/full/r2015a/8.5 matlab/full/r2016b/9.1 matlab/full/r2018a/9.4 matlab/full/r2019b/9.7
+matlab/full/r2015b/8.6 matlab/full/r2017a/9.2 matlab/full/r2018b/9.5
+```
+{: .output}
+
+In this case, there are seven different versions.
+How do we load each copy, and which copy is the default?
+
+Sometimes, on some systems, a module might have a `(default)` next to it.
+This indicates that it is the default (i.e. which would be loaded if we type `module load matlab`).
+In this case, we don't see this, so we will have to load matlab and see what we get.
+
+```
+{{ site.remote.prompt }} module load matlab
+```
+{: .bash}
+
+```
+matlab/full/r2019b/9.7(99):ERROR:151: Module 'matlab/full/r2019b/9.7' depends on one of the module(s) 'gcc-libs/4.9.2'
+matlab/full/r2019b/9.7(99):ERROR:102: Tcl command execution failed: prereq gcc-libs
+```
+{: .output}
+
+Here, we see that the default version of Matlab on the system is r2019b/9.7, which 
+in this case is the most recent version. However, you should not assume that the
+default version is necessarily the latest.
+
+As we saw in the earlier example, there are one or more dependencies.
+
+Suppose we decide to load an earlier version of Matlab, e.g. r2017a/9.2.
+
+```
+{{ site.remote.prompt }} module purge
```

Comment:
Since the goal is to load an earlier version of Matlab, why not `module unload matlab` instead of blowing everything away with `module purge`?

Reply:
I believe we put `purge` here as we want to show how to load all the dependencies, and this ensures that they are not loaded. In fact, the `load` from before doesn't load anything as it fails to find the dependencies. We show `unload` below.
@mattgillucl what do you think?

Reply:
Makes sense, carry on.

Reply:
Yes @dpshelio I wanted to start with a "clean slate", and to show how to achieve that.
We could add a line that one could do `module unload matlab` if one only wanted to unload all Matlab modules.